plasma_framework/docs/design/ald-exit-game-definitions-and-properties.md

```diff
@@ -0,0 +1,133 @@
+# ALD Exit Game Definitions And Properties
+
+## New definitions from ALD (abstract layer design)
+Based on [MoreVP document](https://github.com/omisego/elixir-omg/blob/master/docs/morevp.md). This section adds some new definitions introduced from ALD or give explicit names to some of the implicit characteristics of the origin MoreVP spec.
+
+### SpendingCondition Fulfilled
+1. Checks an input tx and the spending tx, whether the spending tx fulfil the requirement of spending the output of input tx or not.
+1. Basic checks:
+    - Spending tx input points to the output of input tx.
+    - Spending tx fulfil the “lock” condition of the output
+1. This checks on (at most) tx level, does not check things on Plasma protocol level.
+
+### Tx Standard Finalization
+1. MVP:
+    - Tx is in a plasma block. (Checks inclusion proof of plasma block)
+    - It is confirmed by user. (Checks confirm signature is valid for the inclusion block)
+2. MoreVP:
+    - Tx is in a plasma block. (Checks inclusion proof of plasma block)
+
+### Tx Protocol Finalized
+This is the minimal finalization requirement according to each protocol.
+
+1. MoreVp tx: the transaction exists
+2. MVP tx: the transaction is standard finalized
+
+## Exit Properties
+
+### General Properties
+
+#### Valid Spending Tx:
+1. Needs to be protocol finalized
+1. Fulfils the spending condition of the output
+
+#### Input/Output finalized
+1. The input/output has been withdrawn/processed.
+1. Or The output has been used as an input of a processed IFE.
+
+#### Input/Output withdrawable
+The input/output is not finalized already
+
+### In-flight Exit
+
+#### InFlight exit to be exitable
+1. Inputs of the exiting tx should be Standard Finalized (Protocol finalized is not enough)
+1. The state transition of the tx is valid
+1. All inputs fulfill the spending condition with the exiting tx
+1. The transaction itself is Protocol finalized
+
+#### Competing tx
+1. Needs to be protocol finalized
+1. Fulfils the spending condition of an input of the InFlight exiting tx
+
+#### Canonical tx
+1. The tx is exitable
+1. No competing tx that has higher or same priority (txPos) then the exiting one. (PS, same priority is for the case both txs are in-flight)
+1. All inputs of the tx are not finalized
+
+#### Piggybacked input/output to be valid
+1. It is the input or output of the exiting tx
+1. There is no other valid spending tx of the piggybacked input/output existing.
+
+#### Spending Tx of a piggybacked input/output to be valid:
+1. Needs to be Protocol finalized
+1. Fulfils the spending condition of the output that is exiting.
+
+### Standard Exit
+
+#### Standard exit to be exitable
+1. The tx of the output is standard finalized
+1. The exiting output belongs to the tx it claims
```

Comment:
What does that mean additionally to being standard finalized?

Reply:
oh, just want to be a bit more clear, since "standard finalized" is on the tx level not output level. So second line is just saying the output is of the tx.